src/Toolkit/Toolkit.WinUI/Internal/HtmlToView.WinUI.cs

```diff
@@ -169,6 +169,33 @@ private static FrameworkElement CreateBlock(MarkupNode node, EventHandler<Uri>?
                     imageElement.Source = imageSource;
                 return imageElement;
 
+            case MarkupType.Audio:
+            case MarkupType.Video:
+                // Find the first valid <source> child, or use the node's Content
+                string? mediaSrc = node.Content;
+                string? mediaType = null;
+                foreach (var child in node.Children)
+                {
+                    if (child.Type is MarkupType.Source && !string.IsNullOrEmpty(child.Content))
+                    {
+                        mediaSrc = child.Content;
+                        var attr = HtmlUtility.ParseAttributes(child.Token?.Attributes);
+                        attr.TryGetValue("type", out mediaType);
+                        break;
+                    }
+                }
+                if (!string.IsNullOrEmpty(mediaSrc))
+                {
+                    var mediaPlayerElement = new MediaPlayerElement
+                    {
+                        Source = Windows.Media.Core.MediaSource.CreateFromUri(new Uri(mediaSrc, UriKind.RelativeOrAbsolute)),
+                        AreTransportControlsEnabled = true,
+                        Stretch = Stretch.Uniform,
+                    };
+                    return mediaPlayerElement;
+                }
+                return new TextBlock { Text = "Media not available" };
```

Comment:
HTML audio and video tags are usually authored with _fallback content_.  It is arbitrary HTML inside the audio/video tag that should be displayed when media cannot be played/loaded.  We should display it here if possible.

For example: if the node has any non-Source children, render them (might be able to just do `goto case MarkupType.Block`).  And only use a TextBlock as a last resort when there's neither media to play, nor fallback content to show.